packaging/run_inference.py

```diff
@@ -0,0 +1,225 @@
+        os.makedirs(args.path_out, exist_ok=True)
+
+    if args.path_dataset is not None and args.path_images is not None:
+        raise ValueError('You can only specify either --path-dataset or --path-images (not both). See --help for more info.')
+    
+    if args.path_dataset is not None:
+        print('Found a dataset folder. Running inference on the whole dataset...')
+
+        # NOTE: nnUNet only wants the _0000 suffix for files contained in a folder (i.e. when inference is run on a whole dataset)
+        # hence, we create a temporary folder with the proper filenames and delete it after inference is done
+        # More info about that naming convention here: 
+        # https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/dataset_format_inference.md
+        
+        print('Creating temporary folder with proper filenames...')
+        path_data_tmp = convert_filenames_to_nnunet_format(args.path_dataset)
+        path_out = args.path_out
+
+    elif args.path_images is not None:
+        # NOTE: for individual images, the _0000 suffix is not needed. BUT, the images should be in a list of lists
+        # get list of images from input argument
+        print(f'Found {len(args.path_images)} images. Running inference on them...')
+        # path_data_tmp = [[os.path.basename(f)] for f in args.path_images]
+        path_data_tmp = [[f] for f in args.path_images]
+        print(path_data_tmp)
+
+        path_out = args.path_out
+        # # add suffix '_pred' to predicted images
+        # for f in args.path_images:
+        #     path_pred = os.path.join(args.path_out, add_suffix(f, '_pred')) 
+        #     path_out.append(path_pred)
+
+    # uses all the folds available in the model folder by default
+    folds_avail = [int(f.split('_')[-1]) for f in os.listdir(args.path_model) if f.startswith('fold_')]
+
+    # ---------------------------------------------------------------
+    # OPTION 1: Currently, pip install nnUNetv2 does not have the latest version of nnUNet's inference 
+    # which is defined in OPTION 2. Hence, this method
+    # ---------------------------------------------------------------
+
+    print('Starting inference...')
+    start = time.time()
+    # directly call the predict function
+    predictor(
+        list_of_lists_or_source_folder=path_data_tmp, 
+        output_folder=path_out,
+        model_training_output_dir=args.path_model,
+        use_folds=folds_avail,
+        tile_step_size=0.5,
+        use_gaussian=True,                                      # applies gaussian noise and gaussian blur
+        use_mirroring=True if args.use_mirroring else False,    # test time augmentation by mirroring on all axes
+        perform_everything_on_gpu=True if args.use_gpu else False,
+        device=torch.device('cuda', 0) if args.use_gpu else torch.device('cpu'),
+        verbose=False,
+        save_probabilities=False,
+        overwrite=True,
+        checkpoint_name='checkpoint_final.pth',
```

Comment:
Since I kille the training, I had to change this line to:

```
checkpoint_name='checkpoint_best.pth'
```

We could add a condition to check this.

Reply:
Added the condition here! https://github.com/ivadomed/model_seg_sci/pull/40/commits/c302c816d31275b24d87ce0b2b776bc714d7f6cf